src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -587,17 +642,90 @@ def docKeyUpdate(self, msOutDoc, **kwargs):
         for key, value in kwargs.items():
             try:
                 msOutDoc.setKey(key, value)
+                msOutDoc.updateTime()
             except Exception as ex:
                 msg = "Cannot update key {} for doc: {}\n".format(key, msOutDoc['_id'])
                 msg += "Error: {}".format(str(ex))
                 self.logger.warning(msg)
         return msOutDoc
 
-    def docInfoUpdate(self, msOutDoc):
+    def docInfoUpdate(self, msOutDoc, pipeLine=None):
         """
         A function intended to fetch and fill into the document all the needed
         additional information like campaignOutputMap etc.
         """
+
+        # Fill the destinationOutputMap first
+        if msOutDoc['isRelVal']:
+            destinationOutputMap = []
+            wflowDstSet = set()
+            updateDict = {}
+            for dataset in msOutDoc['OutputDatasets']:
+                _, dsn, procString, dataTier = dataset.split('/')
+                destination = set()
+                if dataTier != "RECO" and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                if dataTier == "GEN-SIM":
+                    destination.add('T1_US_FNAL_Disk')
+                if dataTier == "GEN-SIM-DIGI-RAW":
+                    destination.add('T1_US_FNAL_Disk')
+                if dataTier == "GEN-SIM-RECO":
+                    destination.add('T1_US_FNAL_Disk')
+                if "RelValTTBar" in dsn and "TkAlMinBias" in procString and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                if "MinimumBias" in dsn and "SiStripCalMinBias" in procString and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                dMap = {'datasets': [dataset],
+                        'destination': list(destination)}
+                destinationOutputMap.append(dMap)
+                wflowDstSet |= destination
+
+            # here we try to aggregate the destination map per destination
+            aggDstMap = []
+
+            # populate the first element in the aggregated list
+            aggDstMap.append(destinationOutputMap.pop())
```

Comment:
Way too much complicated way to aggregate destinations/datasets. One could use a dict with frozenset keys. Once done, we could then expand that to a list and be done with it (like what I did to the campaign/output map).

Reply:
Need to admit - never tried it before.